sympy/tensor/array/ndim_array.py

```diff
@@ -129,10 +129,18 @@ def f(pointer):
     @classmethod
     def _handle_ndarray_creation_inputs(cls, iterable=None, shape=None, **kwargs):
         from sympy.matrices.matrices import MatrixBase
+        from sympy.tensor.array import SparseNDimArray
+        from sympy import Dict
 
         if shape is None and iterable is None:
             shape = ()
             iterable = ()
+        # Construction of a sparse array from a disctionary
+        elif shape is not None and isinstance(iterable, (dict, Dict)):
```

Comment:
elif -> if as there is a return under it

Reply:
Is it a convention in SymPy?

Reply:
Is this a convention in SymPy?

Reply:
It's not about convention, it's just about avoiding the things which aren't needed. I don't know if you follow some other principles.

Reply:
I changed the assignment of values as other blocs, so problem solved. 
But just out of curiousity, would you please explain to me why should elif -> if? I don't really see the point. As far as I know, a reasonable if-elif-else bloc ensures the logical consistency so that the code is more readable.

Reply:
No issues if you follow other principles. Everyone has their own coding style. It appeared non-intuitive to me to have an if just after an elif.